main.cpp

```diff
@@ -1,95 +1,570 @@
  */
-
+struct ShelterOfCats
+{
+    //1)number of cats
+    int numberOfCats = 5;
+    //2)distance
+    float distance = 1.2f;
+    //3)amount of money to adopt a cat
+    int amountOfMoneyToAdoptACat = 80;
+    //4)age of a cat
+    float ageOfACat = 0.5f;
+    //5)activity level
+    int activityLevel = 3;
+    //1)adopt a cat
+    void adoptACat(int numberOfCat);
+    //2)donate money
+    int donateMoney(double donation);
+    //3)visit the shelter on site
+    void visitTheShelterOnSite(std::string address);
+};
 /*
- 4)
+4)TV series
+5 properties:
+    1)number of season
+    2)number of episode
+    3)duration of one episode
+    4)number of languages supported
+    5)rating
+3 things it can do
+    1)adjust the timeline
+    2)add to watchlist
+    3)watch trailer
  */
-
+struct TvSeries
+{
+    //1)number of season
+    int numberOfSeason = 2;
+    //2)number of episode
+    int numberOfEpisode = 13;
+    //3)length of one episode
+    int lengthOfOneEpisode = 45;
+    //4)number of languages supported
+    int numberOfLanguagesSupported = 3;
+    //5)rating
+    float rating = 7.7f;
+    //1)adjust the timeline
+    void adjustTheTimeline(double currentTime);
+    //2)add to watchlist
+    void addToWatchlist(std::string addressOfTvSeries);
+    //3)watch trailer
+    void watchTrailer(std::string addressOfTrailer);
+};
 /*
- 5)
+5)User
+5 properties:
+    1)user registration time
+    2)type of membership
+    3)gender of user
+    4)level of user
+    5)number of messages
+3 things it can do:
+    1)send messages to other users
+    2)create a playlist   
+    3)Set playlist privacy
  */
-
+struct User
+{
+    //1)user registration days
+    int userRegistrationDays = 344;
+    //2)type of membership
+    int typeOfMemvership = 2;
+    //3)gender of user
+    int genderOfUser = 1;
+    //4)level of user
+    int levelOfUser = 3;
+    //5)number of messages
+    int numberOfMessages = 4;
+    //1)send messages to other users
+    void sendMessagesToOtherUsers(std::string nameOfOtherUser);
+    //2)create a playlist
+    void createAPlaylist(std::string nameOfPlaylist);   
+    //3)Set playlist privacy
+    void setPlaylistPrivacy(int numberOfPlaylist);
+};
 /*
- 6)
+6) Artist
+5 properties:
+    1)age
+    2)number of social updates
+    3)number of followers   
+    4)number of albums
+    5)number of awards
+3 things it can do:
+    1)follow a artist
+    2)enter the group of a artist
+    3)check the hottest song of artist
  */
-
+struct Artist
+{
+    //1)age
+    int age = 23;
+    //2)number of social updates
+    int numberOfSocialUpdates = 14;
+    //3)number of followers
+    int numberOfFollowers = 1002;
+    //4)number of albums
+    int numberOfAlbums = 3;
+    //5)number of awards
+    int numberOfAwards = 5;
+    struct Group
+    {
+        int numberOfMembers = 233;
+        std::string administrator = "sodabuddy";
+        std::string notice = "New album will be released in few weeks";
+    };
```

Comment:
Nested structs need to have 5 properties (member variables) and 3 actions (functions)
- [ ] revise